### PR TITLE
mesh_to_model GH Component

### PR DESCRIPTION
**Pull request by obucklin**
New Gh component that generates CT_Beams from mesh edges. beam z-direction is given by the average of the normals of the adjacent mesh faces. 
### What type of change is this?

- [ ] Bug fix in a **backwards-compatible** manner.
- [x] New feature in a **backwards-compatible** manner.
- [ ] Breaking change: bug fix or new feature that involve incompatible API changes.
- [ ] Other (e.g. doc update, configuration, etc)

### Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I added a line to the `CHANGELOG.md` file in the `Unreleased` section under the most fitting heading (e.g. `Added`, `Changed`, `Removed`).
- [x] I ran all tests on my computer and it's all green (i.e. `invoke test`).
- [x] I ran lint on my computer and there are no errors (i.e. `invoke lint`).
- [ ] I added new functions/classes and made them available on a second-level import, e.g. `compas_timber.datastructures.Beam`.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: src/compas_timber/ghpython/components/CT_Beams_From_Mesh/icon.png
Comment (chenkasirer) [inline src/compas_timber/ghpython/components/CT_Beams_From_Mesh/icon.png:1]: isnt' this  icon already used somewhere?

File: src/compas_timber/ghpython/components/CT_Beams_From_Mesh/metadata.json [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    "category": "COMPAS Timber",
+    "subcategory": "Design",
+    "description": "Creates Beams and joint rules from a planar brep surface.",
+    "exposure": 2,
+    "ghpython": {
+        "isAdvancedMode": true,
+        "iconDisplay": 0,
+        "inputParameters": [
+            {
+                "name": "mesh",
```
Comment (chenkasirer) [inline src/compas_timber/ghpython/components/CT_Beams_From_Mesh/metadata.json:13]: Capitalize argument names to stay consistent with the other components

File: src/compas_timber/ghpython/components/CT_Beams_From_Mesh/code.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+from compas.scene import Scene
+from ghpythonlib.componentbase import executingcomponent as component
+from Grasshopper import DataTree
+from Grasshopper.Kernel.Data import GH_Path
+from Grasshopper.Kernel.GH_RuntimeMessageLevel import Warning
+
+from compas_timber.elements import Beam as CTBeam
+
+
+class Beam_tree_from_mesh(component):
```
Comment (chenkasirer) [inline src/compas_timber/ghpython/components/CT_Beams_From_Mesh/code.py:15]: class name -> `CamelCase`

File: src/compas_timber/ghpython/components/CT_Beams_From_Mesh/code.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+import rhinoscriptsyntax as rs
+from compas.geometry import Line
+from compas.geometry import Vector
+from compas.scene import Scene
+from ghpythonlib.componentbase import executingcomponent as component
+from Grasshopper import DataTree
+from Grasshopper.Kernel.Data import GH_Path
+from Grasshopper.Kernel.GH_RuntimeMessageLevel import Warning
+
+from compas_timber.elements import Beam as CTBeam
```
Comment (chenkasirer) [inline src/compas_timber/ghpython/components/CT_Beams_From_Mesh/code.py:12]: why the renaming?

File: src/compas_timber/ghpython/components/CT_Beams_From_Mesh/icon.png
Comment (obucklin) [reply src/compas_timber/ghpython/components/CT_Beams_From_Mesh/icon.png:1]: yeah I haven't added new icons

File: src/compas_timber/ghpython/components/CT_Beams_From_Mesh/code.py (hunk 3 above)
Comment (obucklin) [reply src/compas_timber/ghpython/components/CT_Beams_From_Mesh/code.py:12]: it is like this in `Beam_from_curve` component...

**Review by chenkasirer — APPROVED** [review]
LGTM

**Pull request merged**